src/components/grid_popover/grid_popover.ts

```diff
@@ -9,13 +10,32 @@ interface Props {
   onClosePopover: () => void;
   onMouseWheel: (ev: WheelEvent) => void;
 }
+
 export class GridPopover extends Component<Props, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-GridPopover";
   static components = { Popover };
 
+  menuService = useState<MenuService>(this.env.menuService);
+  private wasMenuOpen = this.menuService.hasOpenMenu();
+
+  setup() {
+    onPatched(() => {
+      const isMenuOpen = this.menuService.hasOpenMenu();
+      const hasPersistentPopover = this.env.model.getters.hasOpenedPopover();
+      if (isMenuOpen && hasPersistentPopover) {
+        if (!this.wasMenuOpen) {
+          this.env.model.dispatch("CLOSE_CELL_POPOVER");
+        } else {
+          this.menuService.closeActiveMenu();
+        }
+      }
+      this.wasMenuOpen = isMenuOpen;
+    });
```

Comment:
Maybe the persistent popover should be also registered in the menu service instead of having interaction between them and the menuService here ?

The main drawback is that the typing of the menuService would becomes more complex, and it would be harder to use it.